app/controllers/books_controller.rb

```diff
@@ -0,0 +1,53 @@
+class BooksController < ApplicationController
+  def index
+    @books = Book.all
+  end
+
+  def create
+    @book = Book.new(post_params)
+
+    if @book.save
+      # render success in Jbuilder
+    else
+      render json: { message: "400 Bad Request" }, status: :bad_request
+    end
+  end
+
+  def show
+    @book = Book.find_by_id(params[:id])
+
+    if @book.nil?
```

Comment:
You can put this validation in a `before_action` method, so that you don't need to duplicate codes